src/utils/viewer/viewer.cpp

```diff
@@ -552,11 +588,90 @@ void Viewer::viewportEvent(ViewportEvent& event) {
   renderCamera_->setViewport(event.windowSize());
   imgui_.relayout(Mn::Vector2{event.windowSize()} / event.dpiScaling(),
                   event.windowSize(), event.framebufferSize());
+
+  /* Recreate object ID reading renderbuffers that depend on viewport size */
+  selectionDepth_ = Mn::GL::Renderbuffer{};
+  selectionDepth_.setStorage(Mn::GL::RenderbufferFormat::DepthComponent24,
+                             event.framebufferSize());
+  selectionDrawableId_ = Mn::GL::Renderbuffer{};
+  selectionDrawableId_.setStorage(Mn::GL::RenderbufferFormat::R32UI,
+                                  event.framebufferSize());
+  selectionFramebuffer_
+      .attachRenderbuffer(Mn::GL::Framebuffer::BufferAttachment::Depth,
+                          selectionDepth_)
+      .attachRenderbuffer(Mn::GL::Framebuffer::ColorAttachment{1},
+                          selectionDrawableId_)
+      .setViewport({{}, event.framebufferSize()});
 }
 
 void Viewer::mousePressEvent(MouseEvent& event) {
+  if (event.button() == MouseEvent::Button::Right && objectSelectionOn_) {
+    selectionFramebuffer_.bind();
+    selectionFramebuffer_
+        .mapForDraw({{Mn::Shaders::Generic3D::ColorOutput,
+                      Mn::GL::Framebuffer::DrawAttachment::None},
+                     {Mn::Shaders::Generic3D::ObjectIdOutput,
+                      Mn::GL::Framebuffer::ColorAttachment{1}}})
+        .clearDepth(1.0f)
+        .clearColor(1, Mn::Vector4ui{0xffff});
+    CORRADE_INTERNAL_ASSERT(
+        selectionFramebuffer_.checkStatus(Mn::GL::FramebufferTarget::Draw) ==
+        Mn::GL::Framebuffer::Status::Complete);
+
+    int DEFAULT_SCENE = 0;
+    auto& sceneGraph = sceneManager_.getSceneGraph(sceneID_[DEFAULT_SCENE]);
+
+    // if there is a selected object, remove it first
+    if (selectedDrawableId_ >= 0) {
+      for (std::pair<std::string, esp::gfx::DrawableGroup>&& it :
+           sceneGraph.getDrawableGroups()) {
+        esp::gfx::Drawable* d = it.second.getDrawable(selectedDrawableId_);
+        if (d != nullptr) {
+          d->setSelected(false);
+          break;  // the selected object is found and processed
+        }
+      }
+      selectedDrawableId_ = -1;
+    }
```

Comment:
why not store a drawable pointer instead to avoid the lookup?

Reply:
I agree with you.
Long story behind this non-pointer choice. 
Previously people have concerns when using raw pointers. They thought it was unsafe and leading to potential memory leaks or seg-fault, which I did not agree, but I chose to follow such redundant solution to address their concerns.

Reply:
I will use the raw pointers from now on whenever it is proper.